### ISSUE TITLE
Version 22.6.0 fails on MacOS M1 systems

### ISSUE THREAD

**Issue by joseph-wortmann** [psf/black#3231]
Running the latest version of black on MacOS M1 systems fails with the following error:

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/bin/black", line 5, in <module>
    from black import patched_main
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/black/__init__.cpython-39-darwin.so, 0x0002): tried: '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/black/__init__.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e)))
```

The previous version (22.3.0) works correctly.

**Comment by ichard26** [comment]
Hmm interesting. I haven't changed anything about the build configuration or toolchain between 22.3 and 22.6. We have wheels for arm64 and universal2 (ie. code for both x86_64 and arm64 in a single wheel) so it should just work. 
![image](https://user-images.githubusercontent.com/63936253/185455409-68f8d649-8653-464b-8870-c5e5ce46b00e.png)

Unless arm64e is a different architecture than arm64 from a packaging perspective...

cc @cooperlees for ARM knowledge.

**Comment by ichard26** [comment]
The only change I can think of is that something on the host GHA macOS runner (which is mostly out of our control) got upgraded and broke something, but I have no clue.

**Comment by ichard26** [comment]
Oh and if you need to get 22.6.0 working, you can install Black from source via `pip install black --no-binary=black` avoiding all mypyc related shenanigans. Could also download the pure wheel from PyPI, but installing from source should be easier.

**Comment by joseph-wortmann** [comment]
For now I just downgraded. I just thought you should know about the M1 error.

**Comment by cooperlees** [comment]
Howdy - I can not reproduce your error. Can you share how you install please?

```console
crl-m1:~ cooper$ uname -m
arm64
crl-m1:~ cooper$ python3.10 -V
Python 3.10.6
crl-m1:~ cooper$ python3.10 -m venv /tmp/test_black --upgrade-deps
Requirement already satisfied: pip in /private/tmp/test_black/lib/python3.10/site-packages (22.2.2)
Requirement already satisfied: setuptools in /private/tmp/test_black/lib/python3.10/site-packages (63.4.3)
Collecting setuptools
  Downloading setuptools-65.1.0-py3-none-any.whl (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 5.0 MB/s eta 0:00:00
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 63.4.3
    Uninstalling setuptools-63.4.3:
      Successfully uninstalled setuptools-63.4.3
Successfully installed setuptools-65.1.0
crl-m1:~ cooper$ /tmp/test_black/bin/pip install black
Collecting black
  Using cached black-22.6.0-cp310-cp310-macosx_11_0_arm64.whl (1.2 MB)
Collecting platformdirs>=2
  Using cached platformdirs-2.5.2-py3-none-any.whl (14 kB)
Collecting click>=8.0.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting pathspec>=0.9.0
  Using cached pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting mypy-extensions>=0.4.3
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Installing collected packages: mypy-extensions, tomli, platformdirs, pathspec, click, black
Successfully installed black-22.6.0 click-8.1.3 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.5.2 tomli-2.0.1
crl-m1:~ cooper$ /tmp/test_black/bin/black --version
black, 22.6.0 (compiled: yes)
Python (CPython) 3.10.6
crl-m1:~ cooper$ /tmp/test_black/bin/black repos/black/
All done! ✨ 🍰 ✨
44 files left unchanged.
crl-m1:~ cooper$
```

**Comment by cooperlees** [comment]
I even tried with 3.9

```console
crl-m1:~ cooper$ python3.9 -V
Python 3.9.13
crl-m1:~ cooper$ python3.9 -m venv --upgrade-deps /tmp/test_black39
Requirement already satisfied: pip in /private/tmp/test_black39/lib/python3.9/site-packages (22.2.2)
Requirement already satisfied: setuptools in /private/tmp/test_black39/lib/python3.9/site-packages (63.4.3)
Collecting setuptools
  Using cached setuptools-65.1.0-py3-none-any.whl (1.2 MB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 63.4.3
    Uninstalling setuptools-63.4.3:
      Successfully uninstalled setuptools-63.4.3
Successfully installed setuptools-65.1.0
crl-m1:~ cooper$ /tmp/test_black39/bin/pip install black
Collecting black
  Using cached black-22.6.0-cp39-cp39-macosx_11_0_arm64.whl (1.2 MB)
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting click>=8.0.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting pathspec>=0.9.0
  Using cached pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Collecting platformdirs>=2
  Using cached platformdirs-2.5.2-py3-none-any.whl (14 kB)
Collecting typing-extensions>=3.10.0.0
  Using cached typing_extensions-4.3.0-py3-none-any.whl (25 kB)
Collecting mypy-extensions>=0.4.3
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Installing collected packages: mypy-extensions, typing-extensions, tomli, platformdirs, pathspec, click, black
Successfully installed black-22.6.0 click-8.1.3 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.5.2 tomli-2.0.1 typing-extensions-4.3.0
crl-m1:~ cooper$ /tmp/test_black39/bin/black --version
black, 22.6.0 (compiled: yes)
Python (CPython) 3.9.13
```

**Comment by joseph-wortmann** [comment]
@cooperlees I hate it when software does this, but I cannot reproduce it now either, and I reproduced it in multiple virtual environments prior to posting the bug. In fact, I just flipped back to the latest release one of the env's that it was happening in yesterday (I had downgraded black there), and it worked as expected.

I apologize for wasting your time. Suggest that we close this unless/until it reappears...
